com.unity.render-pipelines.high-definition/Editor/Material/Decal/DecalProjectorEditor.cs

```diff
@@ -248,110 +338,133 @@ void DrawHandles()
+                Vector3 pivot = Vector3.zero;
+                Vector3 projectedPivot = new Vector3(0, 0, decalProjector.offset.z - .5f * decalProjector.size.z);
+
+                if (isPivotEditMode)
                 {
-                    if (inEditMode)
-                    {
-                        Vector2 size = new Vector2(
-                            (decalProjector.uvScale.x > 100000 || decalProjector.uvScale.x < -100000 ? 0f : 1f / decalProjector.uvScale.x) * decalProjector.size.x,
-                            (decalProjector.uvScale.y > 100000 || decalProjector.uvScale.y < -100000 ? 0f : 1f / decalProjector.uvScale.y) * decalProjector.size.y
-                        );
-                        Vector2 start = (Vector2)projectedPivot - new Vector2(decalProjector.uvBias.x * size.x, decalProjector.uvBias.y * size.y);
-                        Handles.DrawDottedLines(
-                            new Vector3[]
-                            {
-                                start,                              start + new Vector2(size.x, 0),
-                                start + new Vector2(size.x, 0),     start + size,
-                                start + size,                       start + new Vector2(0, size.y),
-                                start + new Vector2(0, size.y),     start
-                            },
-                            5f);
-                    }
+                    Handles.DrawDottedLines(new[] { projectedPivot, pivot }, k_DotLength);
+                }
+                else
+                {
+                    Quaternion arrowRotation = Quaternion.LookRotation(Vector3.down, Vector3.right);
+                    float arrowSize = decalProjector.size.z * 0.25f;
+                    Handles.ArrowHandleCap(0, projectedPivot, Quaternion.identity, arrowSize, EventType.Repaint);
+                }
 
-                    Vector2 halfSize = decalProjector.size * .5f;
-                    Vector2 halfSize2 = new Vector2(halfSize.x, -halfSize.y);
-                    Vector2 center = (Vector2)projectedPivot + halfSize;
-                    Handles.DrawLine(center - halfSize,  center - halfSize2, 3f);
-                    Handles.DrawLine(center - halfSize2, center + halfSize, 3f);
-                    Handles.DrawLine(center + halfSize,  center + halfSize2, 3f);
-                    Handles.DrawLine(center + halfSize2, center - halfSize, 3f);
+                //draw UV and bolder edges
+                using (new Handles.DrawingScope(Matrix4x4.TRS(decalProjector.transform.position + decalProjector.transform.rotation * new Vector3(decalProjector.offset.x, decalProjector.offset.y, decalProjector.offset.z - .5f * decalProjector.size.z), decalProjector.transform.rotation, Vector3.one)))
+                {
+                    Vector2 UVSize = new Vector2(
```

Comment:
Yes but the uvCenter and uvSize are use differently afterward. so it would be returned by the method which leand to not so evident interest in splitting this into a separate method. I'll look at what I can do